command/unixOSCommand.go

```diff
@@ -26,11 +26,15 @@ func (unixOSCommand) CreateDirectory(
 	remotePath pulumi.StringInput,
 	useSudo bool,
 	opts ...pulumi.ResourceOption) (*remote.Command, error) {
+
+	createCmd := fmt.Sprintf("mkdir -p %v", remotePath)
+	deleteCmd := fmt.Sprintf("bash -c 'if [ -z '$(ls -A %v)' ]; then rm -d %v; fi'", remotePath, remotePath)
```

Comment:
Why you need `bash -c`, here?

Reply:
I need to wrap `if ... ` bash logic in a `bash -c` command to support prefixing it with `sudo`. `sudo if ....` does not work:

```bash
> sudo if true ; then echo hello; fi                                                                                                        
zsh: parse error near `then'
> sudo 'if true ; then echo hello; fi'
sudo: if true ; then echo hello; fi: command not found
> sudo bash -c 'if true ; then echo hello; fi'
hello
```

The reason is that `sudo` executes command with `exec` and not via a shell interpreter. There are alternatives of prefixing the command with `-i`, that requires escaping `;` 

```bash
sudo -i if true \; then echo hello\; fi    
```

or `-s`, that requires wrapping the command in `'`

```bash
sudo -s 'if true ; then echo hello; fi'
```

I chose to wrap the command in `bash -c` as it will work both with or without `sudo` prefix, with no further changes

Source:
* https://unix.stackexchange.com/questions/40568/executing-if-statement-from-command-prompt
* [bash manual](https://linux.die.net/man/1/bash)